### PR TITLE
feat: add message type literal to usage stats

### PR DESCRIPTION
**Pull request by carenthomas**
**Please describe the purpose of this pull request.**
Is it to add a new feature? Is it to fix a bug?

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/letta-ai/letta/issues) or [PRs](https://github.com/letta-ai/letta/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

**Additional context**
Add any other context or screenshots about the PR here.

### CODE REVIEW

**Comment by carenthomas** [comment]
previous:
```
data: {"id":"...","date":"...","message_type":"function_return","function_return":"None","status":"success","function_call_id":"call_xHoMu2jCexf8mqlmfaY2xG5e"}
data: [DONE]
data: {'usage': {'completion_tokens': 31, 'prompt_tokens': 2385
, 'total_tokens': 2416, 'step_count': 1}}
```


now:
```
data: {"id":"...","date":"...","message_type":"function_return","function_return":"None","status":"success","function_call_id":"call_xHoMu2jCexf8mqlmfaY2xG5e"}
data: [DONE]
data: {'message_type': 'usage_statistics', 'completion_tokens': 31, 'prompt_tokens': 2385
, 'total_tokens': 2416, 'step_count': 1}
```

**Comment by carenthomas** [comment]
type checker will be fixed by: https://github.com/letta-ai/letta-cloud/pull/415

**Review by cpacker — APPROVED** [review]

File: letta/client/streaming.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def _sse_post(url: str, data: dict, headers: dict) -> Generator[LettaStreamingRe
                             yield ToolCallMessage(**chunk_data)
                         elif "tool_return" in chunk_data:
                             yield ToolReturnMessage(**chunk_data)
-                        elif "usage" in chunk_data:
-                            yield LettaUsageStatistics(**chunk_data["usage"])
+                        elif "step_count" in chunk_data:
```
Comment (cpacker) [inline letta/client/streaming.py:62]: Slightly weird that we're using `step_count` here but that's fine

Comment (carenthomas) [reply letta/client/streaming.py:62]: yeah it was an abritrary choice but if we move to nesting in the future we can change this back

**Pull request merged**